Final: Integrated Postnatal Register

DevSham opened the pull request.

DevSham committed:
redesign the whole form amd add more concepts

ssmusoke commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-INTEGRATED-POSTNATAL-REGISTER.xml:
```diff
@@ -0,0 +1,629 @@
+          formUILocation="patientDashboard.visitActions"
+          formOrder="6"
+          formIcon="icon-medkit"
+          formShowIf="(visit.active || !visit.active) &amp;&amp; patient.person.dead==false"
```
@shamim2019 This form should only be displayed for females

semujju replied:
Done

ssmusoke commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-INTEGRATED-POSTNATAL-REGISTER.xml:
```diff
@@ -0,0 +1,629 @@
+            <submit class="confirm" submitLabel="save" submitStyle="customSubmit"/>
+        </div>
+    </div>
+    <div class="card mt-2">
```
@shamim2019 Global remove all mt-* classes

semujju replied:
Done

ssmusoke commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-INTEGRATED-POSTNATAL-REGISTER.xml:
```diff
@@ -0,0 +1,629 @@
+          formShowIf="(visit.active || !visit.active) &amp;&amp; patient.person.dead==false"
+          formDisplayStyle="Standard"
+          formLabel="Postnatal Visit Form">
+          <!-- &definitionUiResource=ugandaemrpoc:htmlforms/test.xml -->
```
@shamim2019 This is not correct - fix the name of the file being used

semujju replied:
done

ssmusoke commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-INTEGRATED-POSTNATAL-REGISTER.xml:
```diff
@@ -0,0 +1,629 @@
+<htmlform formName="HMIS MCH 008: INTEGRATED POSTNATAL REGISTER"
```
@shamim2019 Do not use capital letters

semujju replied:
done

DevSham committed:
- remove diagnosis label
- remove family planning header
- Add muac codes drop dow for infant
- put the immunization status in oe line
- Add description for ART codes
- Add save button at the bottom
- grey out the date

DevSham committed:
add new changes

DevSham committed:
Add STK result code
Add STK given concept id
Add MBCP cocept id
FSG erollment concept id added
matrnal nutrition concepts
replace INR No with id instead of uuid

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                    </div>
+                    <div class="card mt-4">
+                        <div class="card-header">
+                            Art
```
Write Art in Caps

semujju replied:
Done

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                        <h4>Woman</h4>
+                            <div class="row">
+                                <div class="col-md-6">
+                                    <label>Art Code</label><!--To do. METS needs to add a description for each of these.-->
```
Write Art in Caps

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                            <h4>Partner</h4>
+                            <div class="row">
+                                <div class="col-md-6">
+                                    <label>Art Code</label>
```
Write Art in Caps

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                                </div>
+                            </div>
+                    </div>
+                </div>
```
Add a textbox for other treatments which you missed

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+          formShowIf="(visit.active || !visit.active) &amp;&amp; patient.person.dead==false&amp;&amp; patient.person.gender=='F'&amp;&amp; sessionLocation.uuid=='e9bc61b5-69ff-414b-9cf0-0c22d6dfca2b'"
+          formDisplayStyle="Standard"
+          formLabel="Postnatal Visit Form">
+    <style>
```
Remove internal css

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+    </style>
+
+    <div class="">
+     <div class="row">
```
Indent the inner divs

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+            </div>
+        </div>
+    </div>
+        </div>
```
This should be well indented

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                                            answerLabels="Yes, No"
+                                            style="radio" class="horizontal" />
+                                    </div>
+                                    <div class="col-md-4">
```
Remove comments

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                        <div class="card-body">
+                            <div class="row">
+                                <div class="col-md-2" >
+                                    <label>Drug</label><!--Need a concept id that groups iron, folic and septrine as a dropdown for this drug-->
```
Remove comments

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                                <div class="row">
+                                    <div class="col-md-4">
+                                        <label>Infant Feeding Options</label>
+                                            <obs conceptId="1151"
```
Use this conceptId "99451", it will provides all the codes required here

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                            </div>
+                            <div class="card-body">
+                                <div class="row">
+                                    <div class="col-md-2" >
```
Use a text area for other treatments offered

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                                </div>
+                                <div class="col-md-3">
+                                    <label>Serial No.</label>
+                                        <obs conceptId="1646AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA" />
```
Use IDs not UUIDs

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                                </div>
+                                <div class="col-md-3">
+                                    <label>Timing for PNC Visit</label>
+                                        <obs conceptId="1732AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA" 
```
Use IDs not UUIDs

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                        </div>
+                        <div class="card-body">
+                            <div class="row">
+                                <obs id="no-fp-method" conceptId="dc7620b3-30ab-102d-86b0-7a5022ba4115"
```
Use IDs not UUIDs

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                            <div class="row">
+                                <div class="col-md-6"> 
+                                <label>Status Of The Breast</label>
+                                    <obs conceptId="07c10f5c-17fd-4a7e-8d72-c2252f589da0"
```
Use IDs not UUIDs

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                                </div>
+                                <div class="col-md-6"> 
+                                <label>Status Of The Cervix</label>
+                                    <obs conceptId="d858f8cb-fe9e-4131-8d91-cd9929cc53de"
```
Use IDs not UUIDs

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                            <div class="row">
+                                <div class="col-md-3">
+                                    <label>Weight (Kgs)</label>
+                                        <obs conceptId="5089AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA" />
```
Use IDs not UUIDs

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                                </div>
+                                <div class="col-md-3">
+                                    <label>M.U.A.C(cm)</label>
+                                        <obs conceptId="1343AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA" id="muac-quantitative" />
```
Use IDs not UUIDs

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                                <div class="col-md-3">
+                                    <label>M.U.A.C</label>
+                                        <includeIf velocityTest="$patient.age &lt; 5">
+                                            <obs conceptId="5f86d19d-9546-4466-89c0-6f80c101191b"
```
Use IDs not UUIDs

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                                                answerLabels="Red &lt; 11.5cm, Yellow 11.5cm - &lt;12.5cm,Green &gt;= 12.5cm" id="muac-qualitative" />
+                                        </includeIf>
+                                        <includeIf velocityTest="$patient.age &lt; 10 &amp;&amp; $patient.age &gt;=5">
+                                            <obs conceptId="5f86d19d-9546-4466-89c0-6f80c101191b"
```
Use IDs not UUIDs

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                                                answerLabels="Red &lt; 13.5cm,Yellow 13.5cm - &lt;14.5cm,Green &gt;=14.5cm" id="muac-qualitative" />
+                                        </includeIf>
+                                        <includeIf velocityTest="$patient.age &lt; 18 &amp;&amp; $patient.age &gt;=10">
+                                            <obs conceptId="5f86d19d-9546-4466-89c0-6f80c101191b"
```
Use IDs not UUIDs

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                                        </includeIf>
+
+                                        <includeIf velocityTest="$patient.age &gt;=18">
+                                            <obs conceptId="5f86d19d-9546-4466-89c0-6f80c101191b"
```
Use IDs not UUIDs

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                                </div>
+                                <div class="col-md-3">
+                                    <label>INR No.</label>
+                                        <obs conceptId="b644c29c-9bb0-447e-9f73-2ae89496a709" id="inrNumber" />
```
Use IDs not UUIDs

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                            <div class="row">
+                                <div class="col-md-5">
+                                    <label>eMTCT Code</label>
+                                        <obs conceptId="d5b0394c-424f-41db-bc2f-37180dcdbe74" 
```
Use IDs not UUIDs

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                                </div>
+                                <div class="col-md-4">
+                                    <label>TFV</label>
+                                        <obs conceptId="5d993591-9334-43d9-a208-11b10adfad85"
```
Use IDs not UUIDs

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                            <div class="row">
+                                <div class="col-md-5">
+                                    <label>eMTCT Code</label>
+                                        <obs conceptId="d5b0394c-424f-41db-bc2f-37180dcdbe74" 
```
Use IDs not UUIDs

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                                <div class="col-md-4">
+                                    <label>TFV</label>
+                                        <obs conceptId="5d993591-9334-43d9-a208-11b10adfad85"
+                                            answerConceptIds="1065AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA,1066AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
```
Use IDs not UUIDs

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                            <div class="row">
+                                <div class="col-md-6">
+                                    <label>Art Code</label><!--To do. METS needs to add a description for each of these.-->
+                                        <obs id="arvs-given" conceptId="a615f932-26ee-449c-8e20-e50a15232763"
```
Use IDs not UUIDs

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                            <div class="row">
+                                <div class="col-md-6">
+                                    <label>Art Code</label>
+                                        <obs id="arvs-given" conceptId="a615f932-26ee-449c-8e20-e50a15232763"
```
Use IDs not UUIDs

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                                <div class="row">
+                                    <div class="col-md-4">
+                                        <label>WHO Clinical Stage</label>
+                                            <obs conceptId="dcdff274-30ab-102d-86b0-7a5022ba4115"
```
Use IDs not UUIDs

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                                    </div>
+                                    <div class="col-md-4">
+                                        <label>CD4 Count</label>
+                                            <obs conceptId="dcbcba2c-30ab-102d-86b0-7a5022ba4115" />
```
Use IDs not UUIDs

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                                    </div>
+                                    <div class="col-md-4">
+                                        <label>CD4 Date</label>
+                                            <obs conceptId="159376AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA" allowFutureDates="false" />
```
Use IDs not UUIDs

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                                <div class="row">
+                                    <div class="col-md-3">
+                                        <label>Viral Load Date</label>
+                                            <obs conceptId="0b434cfa-b11c-4d14-aaa2-9aed6ca2da88" allowFutureDates="false"/>
```
Use IDs not UUIDs

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                                    </div>
+                                    <div class="col-md-3">
+                                        <label>Viral Load Results</label>
+                                            <obs conceptId="dca12261-30ab-102d-86b0-7a5022ba4115" 
```
Use IDs not UUIDs

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                            <div class="row">
+                                    <div class="col-md-6">
+                                        <label>Status of the baby</label>
+                                            <obs conceptId="dd8a2ad9-16f6-44db-82d7-87d6eef14886"
```
Use IDs not UUIDs

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                                    </div>
+                                    <div class="col-md-3">
+                                        <label>M.U.A.C(cm)</label>
+                                            <obs conceptId="1343AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA" id="muac-quantitative" />
```
Use IDs not UUIDs

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                                    <div class="col-md-3">
+                                        <label>M.U.A.C</label>
+                                            <includeIf velocityTest="$patient.age &lt; 5">
+                                                <obs conceptId="5f86d19d-9546-4466-89c0-6f80c101191b"
```
Use IDs not UUIDs

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                                                    answerLabels="Red &lt; 11.5cm, Yellow 11.5cm - &lt;12.5cm,Green &gt;= 12.5cm" id="muac-qualitative" />
+                                            </includeIf>
+                                            <includeIf velocityTest="$patient.age &lt; 10 &amp;&amp; $patient.age &gt;=5">
+                                                <obs conceptId="5f86d19d-9546-4466-89c0-6f80c101191b"
```
Use IDs not UUIDs

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                                                    answerLabels="Red &lt; 13.5cm,Yellow 13.5cm - &lt;14.5cm,Green &gt;=14.5cm" id="muac-qualitative" />
+                                            </includeIf>
+                                            <includeIf velocityTest="$patient.age &lt; 18 &amp;&amp; $patient.age &gt;=10">
+                                                <obs conceptId="5f86d19d-9546-4466-89c0-6f80c101191b"
```
Use IDs not UUIDs

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                                            </includeIf>
+
+                                            <includeIf velocityTest="$patient.age &gt;=18">
+                                                <obs conceptId="5f86d19d-9546-4466-89c0-6f80c101191b"
```
Use IDs not UUIDs

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                                    </div>
+                                    <div class="col-md-4">
+                                        <label>Infant Feeding Counseling Done?</label>
+                                            <obs conceptId="5d993591-9334-43d9-a208-11b10adfad85"
```
Use IDs not UUIDs

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                            </div>
+                            <div class="card-body">
+                                <div class="row">
+                                    <obs conceptId="dc918618-30ab-102d-86b0-7a5022ba4115"
```
Use IDs not UUIDs

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                                <div class="row">
+                                    <obs conceptId="dc918618-30ab-102d-86b0-7a5022ba4115"
+                                        answerConceptId="dc8e1be9-30ab-102d-86b0-7a5022ba4115" answerLabel="BCG" class="col-md-2"/>
+                                    <obs conceptId="dc918618-30ab-102d-86b0-7a5022ba4115"
```
Use IDs not UUIDs

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                                        answerConceptId="dc8e1be9-30ab-102d-86b0-7a5022ba4115" answerLabel="BCG" class="col-md-2"/>
+                                    <obs conceptId="dc918618-30ab-102d-86b0-7a5022ba4115"
+                                        answerConceptId="dc883964-30ab-102d-86b0-7a5022ba4115" answerLabel="OPV" class="col-md-2"/>
+                                    <obs conceptId="dc918618-30ab-102d-86b0-7a5022ba4115"
```
Use IDs not UUIDs

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                                    </div>
+                                    <div class="col-md-6">
+                                        <label>Facility Type</label>
+                                            <obs conceptId="2b00ebe6-edfd-4290-8b8d-942554e3c30d"
```
Use IDs not UUIDs

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                                    </div>
+                                    <div class="col-md-6">
+                                        <label>Facility Type</label>
+                                        <obs conceptId="7ac9966e-71a5-4c61-8998-e763fd94536f"
```
Use IDs not UUIDs

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                                    <div class="col-md-6">
+                                        <label>Facility Type</label>
+                                            <obs conceptId="160481"
+                                                answerConceptIds="6abc57f2-3718-42f6-84c4-4f71e921d590,37549919-5934-448e-b110-cc7868e73ff2,9028e51b-0c27-4b72-bde6-fadba72d1396"
```
Use IDs not UUIDs

semujju commented on omod/src/main/webapp/resources/htmlforms/HMIS-MCH-008-IntegratedPostnatalRegister.xml:
```diff
@@ -0,0 +1,707 @@
+                                    <div class="col-md-4">
+                                        <label>Facility Type</label>
+                                        <obs conceptId="164940"
+                                            answerConceptIds="6abc57f2-3718-42f6-84c4-4f71e921d590,37549919-5934-448e-b110-cc7868e73ff2,9028e51b-0c27-4b72-bde6-fadba72d1396"
```
Use IDs not UUIDs

DevSham committed:
-change art to upercase
-remove internal
-replace UUIDs with IDs
-remove comments
-indent inner divs

DevSham committed:
-change art to upercase -remove internal -replace UUIDs with IDs -remove
comments -indent inner divs

semujju reviewed (commented):
Done
